### ISSUE TITLE
Fluids don't work properly when attempting to swim in them.

### ISSUE THREAD

**Issue by woodiertexas** [Alpha-s-Stuff/TinkersConstruct#87]
### Minecraft Version

1.20.1

### Fabric Version

Fabric 0.14.24 and Quilt 0.21.2

### Fabric API Version

FAPI 0.90.7+1.20.1 and QFAPI 7.4.0 + QSL 6.1.2

### Hephaestus Version

1.20.1-3.6.3.232

### Describe your issue

You walk through fluids as if you're walking on land:
![image](https://github.com/Alpha-s-Stuff/TinkersConstruct/assets/17211100/815813f0-2ae1-415f-8c49-2d1bf3d9b18c)

Vanilla water for comparison:
![image](https://github.com/Alpha-s-Stuff/TinkersConstruct/assets/17211100/24724083-9e38-43c3-a191-c9b2d533b731)

### Other mods

**Fabric Install:**
![image](https://github.com/Alpha-s-Stuff/TinkersConstruct/assets/17211100/71ba1754-9b06-4df4-83dc-8e505130b33a)

**Quilt Install:**
![image](https://github.com/Alpha-s-Stuff/TinkersConstruct/assets/17211100/fb637e84-eedb-4bf6-8eaa-1f4736a132cf)

### Tried reproducing with just Tinkers?

Yes

### Performance Enchancers

Sodium

### Searched for known issues?

Searched open issues

**Comment by AcidArmy** [comment]
This has been the exact same behavior since at least 1.12.2. I think this is just a feature.

**Comment by PengyTheGamer** [comment]
i don't think this is a feature, but i think forge 1.18 and 1.16 lacked the proper fluid behaviours so, this is likely just from it being a port of the forge version

**Comment by jshipley** [comment]
The way most mods handle this now is to add their fluids to the minecraft:fluids/water or minecraft:fluids/lava (minecraft:fluid/water or minecraft:fluid/lava after 1.21) tags. This isn't a perfect solution, but it's enough to get Minecraft to handle the fluid physics and entity interactions.
